mailpoet/lib/Automation/Integrations/MailPoet/Actions/SendEmailAction.php

```diff
@@ -167,18 +181,51 @@ public function saveEmailSettings(Step $step, Automation $automation): void {
     $email->setReplyToName($args['reply_to_name'] ?? '');
     $email->setReplyToAddress($args['reply_to_address'] ?? '');
     $email->setGaCampaign($args['ga_campaign'] ?? '');
+    if ($this->automationHasWooCommerceTrigger($automation)) {
+      $this->storeNewsletterOption($email, NewsletterOptionFieldEntity::NAME_GROUP, 'woocommerce');
+    }
+    if ($this->automationHasAbandonedCartTrigger($automation)) {
+      $this->storeNewsletterOption($email, NewsletterOptionFieldEntity::NAME_EVENT, 'woocommerce_abandoned_shopping_cart');
+    }
+    $this->newslettersRepository->persist($email);
     $this->newslettersRepository->flush();
   }
 
-  private function isTransactional(Step $step, Automation $automation): bool {
-    $allSteps = $automation->getSteps();
-
-    $triggers = array_filter(
-      $allSteps,
-      function(Step $step): bool {
-        return $step->getType() === Step::TYPE_TRIGGER;
+  private function storeNewsletterOption(NewsletterEntity $newsletter, string $optionName, string $optionValue): void {
+    $options = $newsletter->getOptions()->toArray();
+    foreach ($options as $option) {
+      if ($option->getName() === $optionName) {
+        $option->setValue($optionValue);
+        return;
       }
-    );
+    }
+
+    $optionField = $this->newsletterOptionFieldsRepository->findOneBy([
+      'name' => $optionName,
+      'newsletterType' => $newsletter->getType(),
+    ]);
+
+    if (!$optionField) {
+      $optionField = new NewsletterOptionFieldEntity();
```

Comment:
Newsletter option fields are not meant to be used like this. The fields tell you "newsletter of type X has options A, B, C", they shouldn't be created dynamically. In fact, they only ever update [in Populator](https://github.com/mailpoet/mailpoet/blob/a9d89ff514ade08f1115b26fe0647549ae085f80/mailpoet/lib/Config/Populator.php#L375).

But what now... should `automation` emails have all the possible option fields that may be needed, depending on how the send email action is used? I'm not sure. I think it would make sense to rethink this for automations, but that could also get too complex. What are the particular fields you're adding used for? The editor somehow adjusts based on them?

Reply:
Yes. It's is the condition on which the abandoned cart block shows up in the editor.

Reply:
@websupporter I think we can use them, but we should set the new fields [in the Populator](https://github.com/mailpoet/mailpoet/blob/a9d89ff514ade08f1115b26fe0647549ae085f80/mailpoet/lib/Config/Populator.php#L375) and here in the subject set only the option values. Otherwise, sending an email after an abandoned cart trigger will have the side effect that suddenly, automation emails will have these new fields. So it's better to define them all in the same place upfront.

Reply:
[1322790](https://github.com/mailpoet/mailpoet/pull/4836/commits/1322790356ed5684cceb57b05161cb15be42830a)

Not sure about the change.
1. I added them to the populator
2. The newsletter gets created via XHR request in edit_newsletter.tsx. I need to create those options there already and decided to "introduce" `group:automation` and `event:automation`. Empty strings did not work
3. The SendEmailAction toggles now between those and the woocommerce specific ones.